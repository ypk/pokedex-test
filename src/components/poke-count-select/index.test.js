import { render, screen, fireEvent, cleanup, within } from "@testing-library/react";

import PokeCountSelect from "./index";

afterEach(cleanup)

const items = [
    10, 20, 50
];

const pokeCountObject = {
    selectedPokeCountValue: 20,
    handlePokeCountValueChange: jest.fn()
};

describe("Testing <PokeCountSelect />", () => {
    describe("renders without items", () => {
        it("should match snapshot", () => {
            const component = render(<PokeCountSelect {...pokeCountObject} />);
            expect(component.asFragment()).toMatchSnapshot();
        });

        describe("rendering dropdown functionality", () => {
            it("renders dropdown", () => {
                render(<PokeCountSelect {...pokeCountObject} />);
                const buttonElement = screen.getByRole("button");
                expect(buttonElement).toBeInTheDocument();
            });

            it("renders correct option passed in", () => {
                render(<PokeCountSelect {...pokeCountObject} />);
                const buttonElement = screen.getByRole("button");
                expect(buttonElement).toHaveTextContent("Displaying 20 Pokemons");
            });

            it("renders correct text for selected dropdown value", () => {
                pokeCountObject.selectedPokeCountValue = 10;
                render(<PokeCountSelect {...pokeCountObject} />);
                const buttonElement = screen.getByRole("button");
                expect(buttonElement).not.toHaveTextContent("Displaying 20 Pokemons");
                expect(buttonElement).toHaveTextContent("Displaying 10 Pokemons");
                pokeCountObject.selectedPokeCountValue = 20;
            });
        });

        describe("checking button click functionality", () => {
            it("renders dropdown items", async () => {
                render(<PokeCountSelect {...pokeCountObject} />);
                const buttonElement = screen.getByRole("button");
                fireEvent.click(buttonElement);
                const listItems = await screen.getAllByRole("listitem");
                expect(listItems).toHaveLength(3);
            });

            it("renders dropdown items correctly", async () => {
                render(<PokeCountSelect {...pokeCountObject} />);
                const buttonElement = screen.getByRole("button");
                fireEvent.click(buttonElement);
                const listItems = await screen.getAllByRole("listitem");
                listItems.forEach((item, index) => {
                    const { queryByText } = within(item);
                    expect(queryByText(items[index])).toBeInTheDocument()
                })
            });
        });
    });
});


