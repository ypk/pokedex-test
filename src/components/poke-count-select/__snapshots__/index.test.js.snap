// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing <PokeCountSelect /> renders without items should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="relative bg-gray-100"
  >
    <div
      class="absolute right-10"
    >
      <button
        class="rounded-xl inline-flex justify-center w-full rounded-md border p-4 md:p-2 hover:rounded-xl focus:rounded-xl bg-yellow-300 text-sm leading-5 font-medium focus:outline-none hover:bg-yellow-400 focus:bg-yellow-400 focus:border-yellow-400 transition ease-in-out"
      >
        <span
          class="mx-1 text-md leading-6 font-bold"
        >
          Displaying 20 Pokemons
        </span>
        <svg
          class="fill-current w-4 h-6 ml-2 group-focus:text-blue-400 group-hover:text-blue-400"
          version="1.1"
          viewBox="0 0 512 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <g>
            <g>
              <path
                d="M506.157,132.386c-7.803-7.819-20.465-7.831-28.285-0.029l-207.73,207.299c-7.799,7.798-20.486,7.797-28.299-0.015L34.128,132.357c-7.819-7.803-20.481-7.79-28.285,0.029c-7.802,7.819-7.789,20.482,0.029,28.284l207.701,207.27c11.701,11.699,27.066,17.547,42.433,17.547c15.358,0,30.719-5.846,42.405-17.533L506.128,160.67C513.946,152.868,513.959,140.205,506.157,132.386z"
              />
            </g>
          </g>
        </svg>
      </button>
    </div>
  </div>
</DocumentFragment>
`;
